src/storage/store/engine_tests.rs

```diff
@@ -1864,4 +1865,100 @@ mod tests {
             false
         ); // Revoked
     }
+
+    #[tokio::test]
+    async fn pro_users_get_10k_casts() {
```

Comment:
Should also add a test that this fails for non pro users, and for pro users before the protocol feature activates